lisa/tools/chrony.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from typing import cast
+
+from retry import retry
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.util import LisaException
+
+from .service import Service
+
+
+class Chrony(Tool):
+    __leap_status_pattern = re.compile(
+        r"([\w\W]*?)Leap status.*:.*Normal$", re.MULTILINE
+    )
+
+    @property
+    def command(self) -> str:
+        return "chronyc"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        package_name = "chrony"
+        posix_os.install_packages(package_name)
+        return self._check_exists()
+
+    def restart(self) -> None:
+        service = self.node.tools[Service]
+        cmd_result = service._restart_service("chrony")
+        if 0 != cmd_result.exit_code:
+            cmd_result = service._restart_service("chronyd")
+        cmd_result.assert_exit_code()
+
+    @retry(exceptions=LisaException, tries=10, delay=2)
```

Comment:
If the retry can succeed quickly, avoid to wait too long. 1 or 0.5 second is better.